src/components/purchase-detail.js

```diff
@@ -175,6 +177,13 @@ class PurchaseDetail extends Component {
     if (prevState.listing.sellerAddress !== sellerAddress) {
       this.loadSeller(sellerAddress)
     }
+
+    // detect route param change and reload data
+    if (address && address !== this.props.purchaseAddress) {
+      this.setState(defaultState)
+
+      this.loadPurchase()
+    }
```

Comment:
IMO this would be cleaner as:
```
componentWillUnmount() {
    this.setState(defaultState)
}
```
This will prevent the previously viewed purchase data from showing for a second before the new data loads, and the purchase data will be loaded on `componentWillMount()`.
I don't think there's a way this page would need to update without the component being unmounted/remounted, correct?